simtools/testing/configuration.py

```diff
@@ -0,0 +1,225 @@
+"""Integration test configuration."""
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+
+import simtools.utils.general as gen
+
+_logger = logging.getLogger(__name__)
+
+
+def get_list_of_test_configurations(config_files, get_test_names=False):
+    """
+    Return list of test configuration dictionaries or test names.
+
+    Read all configuration files for testing.
+    Add "--help" and "--version" calls for all applications.
+
+    Parameters
+    ----------
+    config_files: list
+        List of integration test configuration files.
+    get_test_names: bool
+        If True, return a list of test names instead of a list of configuration dictionaries.
+
+    Returns
+    -------
+    list:
+        list of test names or of configuration dictionaries.
+
+    """
+    _logger.debug(f"Configuration files: {config_files}")
+
+    configs = _read_configs_from_files(config_files)
+
+    # list of all applications
+    # (needs to be sorted for pytest-xdist, see Known Limitations in their website)
+    _applications = sorted({item["APPLICATION"] for item in configs if "APPLICATION" in item})
+    for app in _applications:
+        configs.extend(
+            [
+                {"APPLICATION": app, "TEST_NAME": "auto-help", "CONFIGURATION": {"HELP": True}},
+                {
+                    "APPLICATION": app,
+                    "TEST_NAME": "auto-version",
+                    "CONFIGURATION": {"VERSION": True},
+                },
+                {"APPLICATION": app, "TEST_NAME": "auto-no_config"},
+            ]
+        )
+
+    if get_test_names:
```

Comment:
Suggest to always return both, to avoid calling it twice in `test_application_from_config`.

Reply:
Good point, I've changed the logic accordingly.